httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java

```diff
@@ -333,7 +333,12 @@ private void digestURI(final URI uri, final Charset charset) {
         this.scheme = uri.getScheme();
         this.encodedSchemeSpecificPart = uri.getRawSchemeSpecificPart();
         this.encodedAuthority = uri.getRawAuthority();
-        this.host = uri.getHost();
+        final String uriHost = uri.getHost();
+        // URI.getHost incorrectly returns bracketed (encoded) IPv6 values. Brackets are an
+        // encoding detail of the URI and not part of the host string.
+        this.host = uriHost != null && InetAddressUtils.isIPv6URLBracketedAddress(uriHost)
```

Comment:
Very nice!

Reply:
Question: Should `#getHost()` contain information that is unwraps IPv6 addresses?

Reply:
I don't think so, the ipv6 address as a host value matches the ipv6 pattern which you've pointed out is less ambiguous than the bracketed value (when used in isolation, not within a URI), so there's no reason to match the java `URI.getHost` brackets.

Reply:
Sorry, I guess my comment wasn't clear enough: Should the method's Javadoc explicitly say that it does not correspond to the behavior of `URI#getHost()` as it it will return raw values.

Does this compute for you?

Reply:
Oh yes, that does make sense, I will update the change. Sorry for the misinterpretation.

Reply:
Update has been pushed.